submissions/andrewklmn/task_js_dom/app.js

```diff
@@ -0,0 +1,165 @@
+      location: "Beverly Hills, Hollywood Hills, any Hills, really.",
+      attractionMethod: "How to attract her: What are you, an idiot? Just flash your cash!",
+    },
+  ],
+  activeIndex: 0,
+};
+
+const menuContainer = document.querySelector('.menu-container');
+const menuUl = document.querySelector('.menu');
+const wraper = document.querySelector('.wraper');
+const header = document.querySelector('.header');
+const image = document.querySelector('.picture');
+const description = document.querySelector('.description');
+const locationArea = document.querySelector('.habitat');
+const attractionMethod = document.querySelector('.method');
+
+const getViewWidth = () => Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+const maxAutoHideMenuWidth = 820;
+
+const toggleMenu = (event)=>{
+  
+  event.target.classList.toggle("burger");
+  
+  menuContainer.classList.toggle("opened") ;
+  wraper.classList.toggle("wide");
+
+  event.stopPropagation();
+  return false;
+};
+
+const menuItemClickListener = (event)=>{
+
+  if (event.target.matches('.menu-item')) {
+    state.activeIndex = event.target.parentNode.id;
+    renderPage(state);
+  
+    if (getViewWidth() < maxAutoHideMenuWidth) {
+      document.querySelector('.btn-toogle-menu').click(); 
+    }
+    event.stopPropagation();
+  };
+
+  return false;
+};
+
+const renderMenu = (state)=>{
+  
+  menuUl.innerHTML = "";
+
+  state.items.forEach((item,index)=>{
+    const li = document.createElement("li");   
+    li.id = index;
+
+    const a = document.createElement("a");
+    a.innerHTML = item.menuTitle;
+    a.classList.add("menu-item");
+    a.href = "javascript:void(0)";
+    
+    if (index==state.activeIndex) {
+      a.classList.toggle("active");
+    };
+
+    li.appendChild(a);
+    menuUl.appendChild(li);
+  });
+};
+
+const showTextContent = ()=>{
+  document.body.classList.remove("text-white");
+}
+
+const renderArticle = (content)=> {
+
+  document.body.classList.add("text-white");
+
+  header.innerHTML = content.title;
+  image.src = content.image;
+  image.alt = content.menuTitle + ' image';
+  description.innerHTML = content.description;
+  locationArea.innerHTML = content.location;
+  attractionMethod.innerHTML = content.attractionMethod;
+};
+
+const renderPage = (state) => {
+
+  renderMenu(state);
+  renderArticle(state.items[state.activeIndex]);
+
+};
+
+// start App when DOM is loaded
+document.addEventListener('DOMContentLoaded', (event)=>{
+  
+  let a = document.createElement("a");
+  a.className = "btn-toogle-menu";
+  a.href = "javascript:void(0)";
+  a.addEventListener('click', toggleMenu);
+  header.parentNode.insertBefore(a,header);
+  
+
+  renderPage(state);
+
+  menuUl.addEventListener('click',menuItemClickListener);
+  image.addEventListener('load',showTextContent);
+
+  if (getViewWidth() < maxAutoHideMenuWidth) {
```

Comment:
Maybe it's better to move this check to your styles? You can handle it in the media queries.